### ISSUE TITLE
Scrollbar on navigation doesn't work properly

### ISSUE THREAD

**Issue by danilodecanini** [ReactTraining/react-router#7617]
## Version

**react-router-dom ^v5.2.0**

## Steps to reproduce

TL; DR;

```
The problem is:

If my page is bigger verticaly my scrollbar does't appear, and only appear if I reload the page
```

Hey there, 

I don't know if this is a bug or something that I'm doing wrong, but here is the deal...

I have my App.js file like this:

```
function App() {
  return (
    <AppProvider>
      <BrowserRouter>
        <ReactNotification />
        <Routes />
        <GlobalStyles />
      </BrowserRouter>
    </AppProvider>
  );
}
```

All my routes is inside <Routes /> component like below:

```
import { Switch } from 'react-router-dom';
import Route from './Route';

<Switch>
      <Route path="/" exact isPrivate component={Index} />
      <Route path="/login" component={SignIn} />      
      <Route path="/admin/login" component={SignInAdmin} />
      <Route path="/admin" isAdminRoute component={Admin} />
      <Route path="/admin/result/:id" isAdminRoute component={Result} />
</Switch>
```

All my routes may have a prop callled 'isPrivate or isAdminRoute' and I did an abstraction of Route from 'react-router-dom' that I verify if there is a token inside Local Storage, and depending of the props that I give to my route I check if the user is an admin or an user.

And my abstraction Route is:

```
export default function RouteWrapper({
  component: Component,
  isPrivate,
  isAdminRoute,
  ...rest
}) {
  const isSignedUser = auth.isUserAuthenticated();
  const isSignedAdmin = auth.isAdminAuthenticated();

  if (isPrivate && !isSignedUser) {
    return <Redirect to="/login" />;
  }

  if (isAdminRoute && !isSignedAdmin) {
    return <Redirect to="/admin/login" />;
  }

  return (
    <Route
      {...rest}
      render={(props) => (
        <DefaultLayout>
          <Component {...props} />
        </DefaultLayout>
      )}
    />
  );
}
```

Everything seems to work fine, but on route with path="/admin/result/:id" I have this component Result and this component have a useEffect inside doing a request to my backend as soon as I access the page, inside this component I have a state that I can control if is loading or not and if is not I show all my data from backend.

**### The problem is:** 

If my page is bigger verticaly my scrollbar does't appear, and only appear if I reload the page

## Expected Behavior

If I access the page above using useHistory hook, my page load normally and scrollbar shows normally independently the page size

## Actual Behavior

When access this page above using useHistory hook from react-router-dom and my page is bigger verticaly the scrollbar doesn't apper and only show if I reload the page.

One thing that I did that don't happens this problem is allow forceRefresh on BrowserRouter, but I think it not suppose to work like this
